internal/interfaceadapter/controller/user_controller.go

```diff
@@ -28,12 +29,20 @@ func NewUserController(userUsecase usecase.UserUsecase) UserController {
 
 func (userController *userController) CreateUser(c echo.Context) error {
 	id := c.FormValue("id")
-	if len(id) <= 0 || 8 < len(id) {
+	match, err := regexp.MatchString("[1-9]{2}(T|G)[1-9]{3}", id); 
```

Comment:
student number regexp should be: `[0-9]{2}[A-Z][0-9]{3}`
alphabet can be more limited tho.